eng/templates/common-tasks.yml

```diff
@@ -0,0 +1,162 @@
+    condition: always()
+
+  - task: PowerShell@2
+    displayName: Pack VS Code Extensions
+    inputs:
+      filePath: $(Build.SourcesDirectory)/eng/package/PackVSCodeExtension.ps1
+      arguments: -stableToolVersionNumber $(StableToolVersionNumber) -outDir "$(Build.ArtifactStagingDirectory)\vscode"
+      workingDirectory: "$(Build.SourcesDirectory)/src"
+      pwsh: true
+
+  # This is an output artifact in the official yml.
+  # - task: PublishBuildArtifacts@1
+  #   displayName: Publish VSCode extension artifacts
+  #   inputs:
+  #     pathToPublish: $(Build.ArtifactStagingDirectory)\vscode
+  #     artifactName: vscode
+  #     artifactType: container
+
+  - task: PowerShell@2
+    displayName: Pack NPM package
+    inputs:
+      filePath: $(Build.SourcesDirectory)/eng/package/PackNpmPackage.ps1
+      arguments: -packageVersionNumber $(StableToolVersionNumber) -outDir "$(Build.ArtifactStagingDirectory)\npm"
+      workingDirectory: "$(Build.SourcesDirectory)/src/polyglot-notebooks"
+      pwsh: true
+
+  # This is an output artifact in the official yml.
+  # - task: PublishBuildArtifacts@1
+  #   displayName: Publish NPM package artifacts
+  #   inputs:
+  #     pathToPublish: $(Build.ArtifactStagingDirectory)\npm
+  #     artifactName: npm
+  #     artifactType: container
+
+  - script: del /S $(Build.SourcesDirectory)\artifacts\packages\$(_BuildConfig)\*.symbols.nupkg
+    displayName: Clean symbol packages
+
+  # This is an output artifact in the official yml.
+  # - task: PublishBuildArtifacts@1
+  #   displayName: Publish packages to artifacts container
+  #   inputs:
+  #     pathToPublish: $(Build.SourcesDirectory)\artifacts\packages\$(_BuildConfig)
+  #     artifactName: packages
+  #     artifactType: container
+
+- ${{ if eq(parameters.platform, 'linux') }}:
+  - script: git config --global core.longpaths true
+    displayName: Enable `git clean` to handle long paths
+
+  - task: NodeTool@0
+    displayName: Add NodeJS/npm
+    inputs:
+      versionSpec: $(NodeJSVersion)
+
+  - script: |
+      mkdir -p "$(Build.SourcesDirectory)/artifacts"
+      rm -rf "$(Build.SourcesDirectory)/artifacts/*"
+      cp eng/resources/* "$(Build.SourcesDirectory)/artifacts"
+    displayName: Prevent test directory crawling
+
+  - pwsh: |
+      $testArg = if ($env:SKIPTESTS -ne "true") { "--test" } else { "" }
+      Write-Host "##vso[task.setvariable variable=_TestArgs]$testArg"
+    displayName: Promote variables
+
+  - script: ./eng/cibuild.sh
+      --configuration $(_BuildConfig)
+      --prepareMachine
+      $(_TestArgs)
+    displayName: Build
+    env:
+      POCKETLOGGER_LOG_PATH: $(PocketLoggerLogPath)
+      TRYDOTNET_PACKAGES_PATH: $(TryDotNetPackagesPath)
+      DOTNET_INTERACTIVE_FRONTEND_NAME: CI
+
+  - pwsh: ./test-retry-runner.ps1 -buildConfig $env:BUILDCONFIG
+    displayName: Test / Blame
+    workingDirectory: $(Build.SourcesDirectory)
+    env:
+      BUILDCONFIG: $(_BuildConfig)
```

Comment:
I *think* we need PocketLogger logs in this step (but I still can't find log files in the build output, so they might not be propagating correctly.)

Reply:
gotcha, will slot them in when we confirm.